torchgeo/datasets/seco.py

```diff
@@ -0,0 +1,226 @@
+        self.checksum = checksum
+
+        if download:
+            self._download()
+
+        if not self._check_integrity():
+            raise RuntimeError(
+                "Dataset not found or corrupted. "
+                + "You can use download=True to download it"
+            )
+
+        self.scene_to_patches = defaultdict(list)
+        for root_directory, directories, fns in os.walk(
+            os.path.join(self.root, self.directory_name)
+        ):
+            if len(directories) == 0 and len(fns) > 0:
+                root_directory, patch_name = os.path.split(root_directory)
+                _, scene_name = os.path.split(root_directory)
+                self.scene_to_patches[scene_name].append(patch_name)
+
+        self.scenes = sorted(self.scene_to_patches.keys())
+        for scene_name in self.scenes:
+            self.scene_to_patches[scene_name] = sorted(
+                self.scene_to_patches[scene_name]
+            )
+
+    def __getitem__(self, index: int) -> Dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            sample with an "image" in 5xCxHxW format where the 5 indexes over the same
+                patch sampled from different points in time by the SeCo method
+        """
+        scene_name = self.scenes[index]
+        patch_names = self.scene_to_patches[scene_name]
+
+        imagery = [
+            self._load_patch(scene_name, patch_name) for patch_name in patch_names
+        ]
+
+        sample = {"image": torch.stack(imagery, dim=0)}
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.scenes)
+
+    def _load_patch(self, scene_name: str, patch_name: str) -> Tensor:
+        """Load a single image patch.
+
+        Args:
+            scene_name: the name of the scene to load from, e.g. '019999'
+            patch_name: the name of the patch to load, e.g.
+                '20200713T075609_20200713T081050_T36QZH'
+
+        Returns:
+            the image with the subset of bands specified by ``self.bands``
+        """
+        all_data = []
+        for band in self.bands:
+            fn = os.path.join(
+                self.root,
+                self.directory_name,
+                scene_name,
+                patch_name,
+                f"{band}.tif",
+            )
+            with rasterio.open(fn) as f:
+                band_data = f.read(1)
+                height, width = band_data.shape
+                assert height == width
+                if height < 264 and width < 264:
+                    band_data = cv2.resize(
+                        band_data, (264, 264), interpolation=cv2.INTER_LINEAR
+                    )
```

Comment:
If this is the only thing you're using cv2 for I would use one of our required deps instead. PIL has a resize function too.

Reply:
cv2 is an order of magnitude faster:

![image](https://user-images.githubusercontent.com/1088078/140554848-87ad7617-d8d3-4a0b-a8bb-abb9e0c5967e.png)

Reply:
Put differently, _just_ PIL resize will take 4 hours of processing time per epoch over the 1 mil image version of this dataset, while cv2 will take 20 minutes.

Reply:
actually the difference depends on the image size (PIL gets worse faster with larger image sizes) -- with the sizes of these images the differences is 1/2

Reply:
For completeness

![image](https://user-images.githubusercontent.com/1088078/140564570-00923dab-cba7-44aa-8873-7e6a286c53bc.png)

Reply:
Added a TODO to come back and look for speedups